controller/controller_test.go

```diff
@@ -859,31 +866,288 @@ func TestController_WaitForStatus_Timeout(t *testing.T) {
+			taskErrAssertion: func(err error) {
+				Expect(errgo.Cause(err)).To(Equal(updateNotAllowedError))
+			},
+		},
+		// Test that no growth, a minalive of 1, with one unit, does return an error.
+		{
+			fleetSetUp: func(f *fleetMock) {
+				f.On("GetStatusWithMatcher", mock.AnythingOfType("func(string) bool")).Return(
+					[]fleet.UnitStatus{
+						{
+							Name:    "jabberwocky-unit@1.service",
+							Current: string(StatusRunning),
+							SliceID: "1",
+						},
+					},
+					nil,
+				)
+			},
+			req: Request{
+				RequestConfig: RequestConfig{
+					Group: "jabberwocky",
+				},
+			},
+			opts: UpdateOptions{
+				MinAlive:  1,
+				MaxGrowth: 0,
+				ReadySecs: 1,
+			},
+			taskErrAssertion: func(err error) {
+				Expect(errgo.Cause(err)).To(Equal(updateNotAllowedError))
+			},
+		},
+		// Test that growth of 8, a minalive of 3, with one running unit, does return an error.
+		{
+			fleetSetUp: func(f *fleetMock) {
+				f.On("GetStatusWithMatcher", mock.AnythingOfType("func(string) bool")).Return(
+					[]fleet.UnitStatus{
+						{
+							Name:    "jabberwocky-unit@1.service",
+							Current: string(StatusRunning),
+							SliceID: "1",
+						},
+					},
+					nil,
+				)
+			},
+			req: Request{
+				RequestConfig: RequestConfig{
+					Group: "jabberwocky",
+				},
+			},
+			opts: UpdateOptions{
+				MinAlive:  3,
+				MaxGrowth: 8,
+				ReadySecs: 1,
+			},
+			taskErrAssertion: func(err error) {
+				Expect(errgo.Cause(err)).To(Equal(updateNotAllowedError))
+			},
+		},
+		// Test that a growth of 1, alive of 1, with one units, does not return an error.
+		{
+			fleetSetUp: func(f *fleetMock) {
+				f.On("GetStatusWithMatcher", mock.AnythingOfType("func(string) bool")).Return(
+					[]fleet.UnitStatus{
+						{
+							Name:    "zebra-unit@1.service",
+							Current: string(StatusRunning),
+							SliceID: "1",
+						},
+					},
+					nil,
+				)
+			},
+			req: Request{
+				RequestConfig: RequestConfig{
+					Group: "zebra",
+				},
+			},
+			opts: UpdateOptions{
+				MinAlive:  1,
+				MaxGrowth: 1,
+				ReadySecs: 1,
+			},
+			taskErrAssertion: func(err error) {
+				Expect(err).To(BeNil())
+			},
+			controllerErrAssertion: func(err error) {
+				Expect(err).To(BeNil())
+			},
+		},
+		// Test that no growth, a minalive of 1, with two units, does not return an error.
+		{
+			fleetSetUp: func(f *fleetMock) {
+				f.On("GetStatusWithMatcher", mock.AnythingOfType("func(string) bool")).Return(
+					[]fleet.UnitStatus{
+						{
+							Name:    "antelope-unit@1.service",
+							Current: string(StatusRunning),
+							SliceID: "1",
+						},
+						{
+							Name:    "antelope-unit@2.service",
+							Current: string(StatusRunning),
+							SliceID: "2",
+						},
+					},
+					nil,
+				)
+			},
+			req: Request{
+				RequestConfig: RequestConfig{
+					Group: "antelope",
+				},
+			},
+			opts: UpdateOptions{
+				MinAlive:  1,
+				MaxGrowth: 0,
+				ReadySecs: 1,
+			},
+			taskErrAssertion: func(err error) {
```

Comment:
these look weird - can't we split this up into two tests? One that tests successfull stuff and one that expect errors to happen?